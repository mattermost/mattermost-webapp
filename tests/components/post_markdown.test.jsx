// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import PostMarkdown from 'components/post_markdown/post_markdown';

describe('components/PostMarkdown', () => {
    const baseProps = {
        imageProps: {},
        isRHS: false,
        message: 'message',
        post: {},
    };

    test('should render properly with an empty post', () => {
        const wrapper = shallow(
            <PostMarkdown {...baseProps}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should render properly with a post', () => {
        const props = {
            ...baseProps,
            post: {
                props: {
                    channel_mentions: {
                        test: {
                            display_name: 'Test',
                        },
                    },
                },
            },
        };
        const wrapper = shallow(
            <PostMarkdown {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
