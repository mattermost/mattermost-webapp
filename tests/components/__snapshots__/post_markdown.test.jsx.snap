// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PostMarkdown should render properly with a post 1`] = `
<Connect(Markdown)
  channelNamesMap={
    Object {
      "test": Object {
        "display_name": "Test",
      },
    }
  }
  imageProps={Object {}}
  isRHS={false}
  message="message"
  proxyImages={true}
/>
`;

exports[`components/PostMarkdown should render properly with an empty post 1`] = `
<Connect(Markdown)
  imageProps={Object {}}
  isRHS={false}
  message="message"
  proxyImages={true}
/>
`;
